src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs

```diff
@@ -86,6 +88,45 @@ public AWSResourceQueryer(IAWSClientFactory awsClientFactory)
             _awsClientFactory = awsClientFactory;
         }
 
+        public async Task<List<Subnet>> DescribeSubnets(List<string>? subnetIds = null)
+        {
+            var ec2Client = _awsClientFactory.GetAWSClient<IAmazonEC2>();
+
+            var request = new DescribeSubnetsRequest();
+            if (subnetIds != null)
+                request.SubnetIds = subnetIds;
+
+            return await ec2Client.Paginators
+                .DescribeSubnets(request)
+                .Subnets
+                .ToListAsync();
+        }
+
+        public async Task<List<SecurityGroup>> DescribeSecurityGroups(List<string>? subnetIds = null)
+        {
+            var ec2Client = _awsClientFactory.GetAWSClient<IAmazonEC2>();
+            var request = new DescribeSecurityGroupsRequest();
+            if (subnetIds != null)
```

Comment:
Add a comment about how null means return all security groups and empty collection returns no security groups.

Reply:
Done

Reply:
I changed the function to take in a VPC Id instead of a subnet list